packages/service-library/src/servicelib/rabbitmq/_client.py

```diff
@@ -142,12 +166,21 @@ async def _on_message(
                             if not await message_handler(message.body):
                                 await message.nack()
                     except Exception:  # pylint: disable=broad-exception-caught
-                        _logger.exception(
-                            "unhandled exception when consuming RabbitMQ message, "
-                            "this is catched but should not happen. "
-                            "Please check, message will be queued back!"
-                        )
-                        await message.nack()
+                        if (
+                            max_retries_upon_error is not None
+                            and retry_count < max_retries_upon_error
+                        ):
+                            message.headers[_HEADER_X_RETRY_COUNT] = retry_count + 1
+                            _logger.exception(
+                                _get_unhandled_exception_message(
```

Comment:
why don't you just create a message instead of a function? this creates unnecessary performance issues right?
what about
```python
msg = "unhandled exception when consuming RabbitMQ message, "
                            "this is catched but should not happen. "
                            "Please check, message will be queued back!"
if (
                            max_retries_upon_error is not None
                            and retry_count < max_retries_upon_error
                        ):
  message.headers[_HEADER_X_RETRY_COUNT] = retry_count + 1
  msg = f"{msg} {(attempt {message.headers[_HEADER_X_RETRY_COUNT]}/{max_retries_upon_error})}"

```

Reply:
I have refactored this and should look way better now